app/src/main/java/com/termux/x11/input/TouchInputHandler.java

```diff
@@ -361,6 +397,35 @@ private class GestureListener extends GestureDetector.SimpleOnGestureListener
         public boolean onScroll(MotionEvent e1, MotionEvent e2, float distanceX, float distanceY) {
             int pointerCount = e2.getPointerCount();
 
+            // For captured touchpad pointer:
+            // Automatic (for touchpad) mode is needed because touchpads ignore screen orientation and report physical X and Y
+            if ((e2.getSource() & InputDevice.SOURCE_TOUCHPAD) == InputDevice.SOURCE_TOUCHPAD
+                    && mInputStrategy instanceof InputStrategyInterface.TrackpadInputStrategy) {
+                float temp;
+                int transform = capturedPointerTransformation;
+                if (capturedPointerTransformation == CapturedPointerTransformation.AUTO) {
+                    for (Display display : mDisplayManager.getDisplays()) {
+                        if (display.getState() == Display.STATE_ON) {
+                            transform = display.getRotation() % 4;
+                            break;
+                        }
+                    }
+                }
```

Comment:
Is there any specific reason to obtain display via `DisplayManager` and not using `mActivity.getLorieView().getDisplay()`? This definitely will break in multi-display mode.

Reply:
Cuz in smarphone HDMI external display DeX mode (using phone as software touchpad), the transformation should depend on phone orientation (=soft touchpad) but `mActivity.getLorieView().getDisplay()` return HDMI display which is useless. 
This enumeration works well for case of both external display DeX + software touchpad and , tablet hardware touchpad. 

I do reckon the risk of not working well in some more complex setup, such as external display DeX + hardware touchpad, or Tablet in external display DeX + hardware touchpad, and perhaps more etc. 

A better approach might be depends on input device information. But I don't have both devices so I used that workaround. FYR. soft touchpad might be "sec_touchpad" but I am not so sure. In case you want to implement: 

![Screenshot from 2024-05-17 08-01-47](https://github.com/termux/termux-x11/assets/46513942/c57774fd-e73e-42f0-b6b1-4e28cb9e7d41)